fix: allow injecting JWT options when creating a VC fixes #89

nikolockenvitz opened the pull request:
I created a proposal to fix the bug described in #89. Happy to hear your feedback.

mirceanis commented on src/types.ts:
```diff
@@ -232,6 +232,12 @@ export interface CreateCredentialOptions {
+  /**
+   * Allows including or overriding some header parameters for the resulting JWT.
+   */
+  jwt?: Partial<JWTOptions>
```
I find this layering of options a little confusing.
Perhaps it would be easier to have:
```ts
export interface CreateCredentialOptions extends Partial<JWTOptions> {
```

And then call `createJWT` with `...options` instead of `...options.jwt`

nikolockenvitz replied:
This would then also inject `options.removeOriginalFields` and `options.header` into the second argument of `createJWT`. Is this a problem?

mirceanis replied:
It's not a problem.
`createJWT` does not use those fields.

mirceanis reviewed (commented):
Please read my suggestion about the options layering.

Also, I guess options should also be propagated when creating presentations too, and then some tests added.

nikolockenvitz commented:
I added a test to check whether the created VC has a correct `exp` set. I am not sure whether a test for `canonicalize` makes sense and how to do that with jest.

I updated the implementation as you suggested and also did the same for the Verifiable Presentation.
There, I was not sure whether a test makes sense because `iat` is explicitly set to `undefined` for VPs.

mirceanis reviewed (approved):
Excellent work!
Thank you for your contribution.
